packages/rulesets/src/spectral/functions/patch-properties-correspond-to-put-properties.ts

```diff
@@ -0,0 +1,65 @@
+// Validates that each patch request body contains one or more properties present in the corresponding put request body,
+// and contains only properties present in the put request body.
+// RPC Code: RPC-Patch-V1-01
+
+import _ from "lodash"
+import { getProperties } from "./utils"
+
+const ERROR_MESSAGE =
+  "A patch request body must only contain properties present in the corresponding put request body, and must contain at least one of the properties."
+const PARAM_IN_BODY = (paramObject: any) => paramObject.in === "body"
+const PATCH = "patch"
+const PUT = "put"
+const PARAMETERS = "parameters"
+
+export const patchPropertiesCorrespondToPutProperties = (pathItem: any, _opts: any, ctx: any) => {
+  if (pathItem === null || typeof pathItem !== "object") {
+    return []
+  }
+
+  const path = ctx.path.concat([PATCH, PARAMETERS])
+  const error = [{ message: ERROR_MESSAGE, path: path }]
+  const errors: any = []
+
+  // array of all the patch body param properties
+  const patchBodyProperties: any[] = pathItem[PATCH]?.parameters?.filter(PARAM_IN_BODY).map((param: any) => getProperties(param.schema))
+  // array of all the put body param properties
+  const putBodyProperties: any[] = pathItem[PUT]?.parameters?.filter(PARAM_IN_BODY).map((param: any) => getProperties(param.schema))
+
+  const patchBodyPropertiesEmpty = patchBodyProperties.length < 1
+  const putBodyPropertiesEmpty = putBodyProperties.length < 1 
+
+  //patch without at least one body properties => error
+  if (patchBodyPropertiesEmpty) {
+    return [
+      {
+        message: "Patch operations body cannot be empty.",
+        path: path,
+      },
+    ]
+  }
+
+  // patch body is empty while put body nonempty
+  // or patch body nonempty while put body empty => error
+  if (patchBodyPropertiesEmpty != putBodyPropertiesEmpty) {
```

Comment:
> if (patchBodyPropertiesEmpty != putBodyPropertiesEmpty) { [](http://example.com/codeflow?start=1&length=58)

Have an explicit condition and error code to check that the patch body is non empty and put is empty